app/src/main/java/at/roteskreuz/stopcorona/model/repositories/InfectionMessengerRepository.kt

```diff
@@ -203,10 +201,23 @@ class InfectionMessengerRepositoryImpl(
         downloadMessagesStateObserver.loading()
         withContext(coroutineContext) {
             try {
-                val archives: List<File> = fetchBatchDiagnosisKeysBasedOnInfectionLevel(warningType)
-                val contextToken = BATCH + "," + warningType + "," + UUID.randomUUID().toString()
+                val warningType = quarantineRepository.getCurrentWarningType()
+                val contextToken = UUID.randomUUID().toString()
+                val index = apiInteractor.getIndexOfDiagnosisKeysArchives()
+                val fullBatchParts = fetchFullBatchDiagnosisKeys(index.fullBatchForWarningType(warningType))
+                val dailyBatchesParts = fetchDailyBatchesDiagnosisKeys(index.dailyBatches)
                 lastScheduledToken = contextToken
-                exposureNotificationRepository.processBatchDiagnosisKeys(archives, contextToken)
+                val fullSession = DbFullSession(
+                    session = DbSession(
+                        token = contextToken,
+                        warningType = warningType.name
+                    ),
+                    fullBatchParts = fullBatchParts,
+                    dailyBatchesParts = listOf(DbDailyBatchPart(token = contextToken, batchNo = 1, intervalStart = 0, path = "/"))
+
+                )
+                sessionDao.insertOrUpdateFullSession(fullSession)
+                // exposureNotificationRepository.processBatchDiagnosisKeys(archives, contextToken)
```

Comment:
put back in

Reply:
it´s put back in and properly assiciated with the processing logic